grails-app/controllers/org/pih/warehouse/shipping/ContainerApiController.groovy

```diff
@@ -54,4 +56,25 @@ class ContainerApiController extends BaseDomainApiController {
         }
     }
 
+    def save = {
+        try {
+            JSONObject jsonObject = request.JSON
+            Shipment shipment = shipmentService.getShipment(jsonObject.getString("shipment.id"))
+            if(!jsonObject.containsKey("container.name") && !jsonObject.containsKey("container.containerNumber")){
+                render shipmentService.initializeContainerFromShipment(shipment) as JSON
+            }else{
+                Container container = new Container(shipment: shipment)
```

Comment:
No need to pass in shipment here because it should be set properly in shipment.addToContainers()